membership-attribute-service/app/services/AttributesFromZuora.scala

```diff
@@ -52,13 +55,24 @@ class AttributesFromZuora(implicit val executionContext: ExecutionContext, syste
           case Failure(e) => SafeLogger.error(scrub"Failed to retrieve attributes from cache for $identityId because $e")
         }
 
+    def userHasDigiSub(accountsWithSubscriptions: List[AccountWithSubscriptions]) =
+      accountsWithSubscriptions.flatMap(_.subscriptions).exists(_.asDigipack.isDefined)
+
     lazy val getAttrFromZuora =
       for {
         accounts <- EitherT(identityIdToAccounts(identityId))
-        subscriptions <- EitherT(if (accounts.size > 0) getSubscriptions(accounts, subscriptionsForAccountId) else Future.successful(\/.right[String, List[AccountWithSubscriptions]](Nil)))
+        subscriptions <- EitherT(
+          if (accounts.size > 0) getSubscriptions(accounts, subscriptionsForAccountId)
+          else Future.successful(\/.right[String, List[AccountWithSubscriptions]](Nil))
+        )
+        giftSubscriptions <- EitherT(
+          if(userHasDigiSub(subscriptions)) Future.successful(\/.right[String, List[GiftSubscriptionsFromIdentityIdRecord]](Nil))
+          else giftSubscriptionsForIdentityId(identityId)
+        )
+        giftAttributes = giftSubscriptions.headOption.map(record => ZuoraAttributes(identityId, DigitalSubscriptionExpiryDate = Some(record.TermEndDate)))
```

Comment:
how do we know that the first one is the latest expiry date?

Reply:
Well we know that it is active because of the query but you're right that it would be good to pick the one with the latest expiry in the event that there is more than one

Reply:
active just means the subscription version is the current version, not the dates are current, 
https://knowledgecenter.zuora.com/Billing/Subscriptions/Subscriptions/A_Viewing_a_Subscription#Active_Status_Subtypes:~:text=If%20you%20are%20using%20the%20Zuora,or%20equal%20to%20the%20current%20date.

Reply:
Ah, good to know, thanks!